Sourcery/Parsing/Composer.swift

```diff
@@ -91,13 +91,74 @@ struct Composer {
         return unique.values.sorted { $0.name < $1.name }
     }
 
+    private func resolveGenerics(typeName: TypeName, unique: [String: Type]) {
+        let genericSplit = typeName.unwrappedTypeName.characters.split(separator: "<", maxSplits: 1)
+
+        guard genericSplit.count == 2 else {
+            return
+        }
+
+        let typeCharacters = genericSplit[0]
+        var genericCharacters = genericSplit[1]
+
+        guard genericCharacters.count > 1, genericCharacters.removeLast() == ">" else {
+            return
+        }
+
+        var genericTypeStrings = [String]()
+
+        var subGenericsCounter = 0
+        var characterBuffer = [Character]()
+
+        for character in genericCharacters {
+            switch character {
+            case "<":
+                characterBuffer.append(character)
+                subGenericsCounter += 1
+            case ">":
+                characterBuffer.append(character)
+                subGenericsCounter -= 1
+            case ",":
+                if subGenericsCounter == 0 {
+                    genericTypeStrings.append(String(characterBuffer).trimmingCharacters(in: [" "]))
+                    characterBuffer = []
+                } else {
+                    characterBuffer.append(character)
+                }
+            default:
+                characterBuffer.append(character)
+            }
+        }
+
+        if !characterBuffer.isEmpty {
+            genericTypeStrings.append(String(characterBuffer).trimmingCharacters(in: [" "]))
+        }
+
+        let genericTypeNames = genericTypeStrings.flatMap { genericTypeName -> TypeName in
+            let typeName = TypeName(genericTypeName)
+            resolveGenerics(typeName: typeName, unique: unique)
+            return typeName
+        }
+
+        let genericTypes = genericTypeStrings.flatMap { genericTypeName in
+            return unique[genericTypeName]
+        }
+
+        typeName.generic = GenericType(name: String(typeCharacters), referencedTypes: genericTypes, referencedTypeNames: genericTypeNames)
+    }
+
     private func resolveType(typeName: TypeName, containingType: Type?, unique: [String: Type], modules: [String: [String: Type]], typealiases: [String: Typealias]) -> Type? {
         let actualTypeName = self.actualTypeName(for: typeName, containingType: containingType, unique: unique, typealiases: typealiases)
         if let actualTypeName = actualTypeName, actualTypeName != typeName.unwrappedTypeName {
             typeName.actualTypeName = TypeName(actualTypeName)
         }
 
         let lookupName = typeName.actualTypeName ?? typeName
+
+        if lookupName.isGeneric, lookupName.generic == nil {
```

Comment:
Then I guess we should do that while parsing array/dictionary, just creating a `GenericType` from information that we parsed already.